src/robot/model/itemlist.py

```diff
@@ -45,6 +46,8 @@ class ItemList(MutableSequence[T]):
     """
 
     __slots__ = ['_item_class', '_common_attrs', '_items']
+    # TypeVar T needs to be applied to a variable to be compatible with @set_type_hints
+    item_type: Type[T] = Any
```

Comment:
- Comment uses old decorator name.
- It might be a good idea to more explicitly mention that `copy_signature` doesn't work without this. There are no automated tests to verify it, so it's easy to break it.
- `Any` is a type and using it as a value here looks a bit odd. The value cannot be `None` or `object()` because they aren't compatible with `Type[T]`, but `type(object)` ought to work. I'm not sure is it any better than `Any`, though. Your call.

Reply:
Since this pattern will be used in `model.Body` in a future PR, I decided to deal with this by making a dedicated type alias `known_at_runtime` which just equals `type(object)`

Reply:
Sounds like a good idea. Since it's a type, using a name like `LaterKnownType`, `TypeKnowAtRunTime`, `TypeType` or something like that could be better.